Analysis/Tasks/PWGHF/HFJpsiCandidateSelector.cxx

```diff
@@ -0,0 +1,257 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file HFJpsiCandidateSelector.cxx
+/// \brief Jpsi selection task.
+/// \author Biao Zhang <biao.zhang@cern.ch>, CERN
+/// \author Nima Zardoshti <nima.zardoshti@cern.ch>, CERN
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "AnalysisDataModel/HFSecondaryVertex.h"
+#include "AnalysisDataModel/HFCandidateSelectionTables.h"
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::aod::hf_cand_prong2;
+
+static const int npTBins = 9;
+static const int nCutVars = 4;
+//temporary until 2D array in configurable is solved - then move to json
+//    mass  dcaxy dcaz  pt_e
+constexpr double cuts[npTBins][nCutVars] =
+  {{0.5, 0.2, 0.4, 1},  /* pt<0.5   */
+   {0.5, 0.2, 0.4, 1},  /* 0.5<pt<1 */
+   {0.5, 0.2, 0.4, 1},  /* 1<pt<2   */
+   {0.5, 0.2, 0.4, 1},  /* 2<pt<3   */
+   {0.5, 0.2, 0.4, 1},  /* 3<pt<4   */
+   {0.5, 0.2, 0.4, 1},  /* 4<pt<5   */
+   {0.5, 0.2, 0.4, 1},  /* 5<pt<7   */
+   {0.5, 0.2, 0.4, 1},  /* 7<pt<10  */
+   {0.5, 0.2, 0.4, 1}}; /* 10<pt<15 */
+
+/// Struct for applying Jpsi selection cuts
+
+struct HFJpsiCandidateSelector {
+
+  Produces<aod::HFSelJpsiCandidate> hfSelJpsiCandidate;
+
+  Configurable<double> d_pTCandMin{"d_pTCandMin", 0., "Lower bound of candidate pT"};
+  Configurable<double> d_pTCandMax{"d_pTCandMax", 50., "Upper bound of candidate pT"};
+
+  Configurable<double> d_pidTPCMinpT{"d_pidTPCMinpT", 0.15, "Lower bound of track pT for TPC PID"};
+  Configurable<double> d_pidTPCMaxpT{"d_pidTPCMaxpT", 10., "Upper bound of track pT for TPC PID"};
+
+  Configurable<double> d_TPCNClsFindablePIDCut{"d_TPCNClsFindablePIDCut", 70., "Lower bound of TPC findable clusters for good PID"};
+  Configurable<double> d_nSigmaTPC{"d_nSigmaTPC", 3., "Nsigma cut on TPC only"};
+
+  /// Gets corresponding pT bin from cut file array
+  /// \param candpT is the pT of the candidate
+  /// \return corresponding bin number of array
+  template <typename T>
+  int getpTBin(T candpT)
```

Comment:
@vkucera @nzardosh. These functions are duplicated in all tasks. Can we just keep them in some common utility class?

Reply:
yes if we can find a god solution for this it would be perfect. The only issue is that one needs to pass the array object to the function if its done centrally, but we can do this by reference.